diffcp/cone_program.py

```diff
@@ -99,10 +105,12 @@ def solve_and_derivative(A, b, c, cone_dict, warm_start=None, **kwargs):
             that applies the adjoint of the derivative of the cone program at
             (A, b, and c) to the perturbations `dx`, `dy`, `ds`, which must be
             NumPy arrays. The output `dA` matches the sparsity pattern of `A`.
-
     Raises:
         SolverError: if the cone program is infeasible or unbounded.
     """
+    if mode not in ["dense", "sparse", "lsqr"]:
+        return NotImplementedError
```

Comment:
nit: This should be a `ValueError`. Something like

```
raise ValueError("Unsupported mode %s; the supported modes are 'dense', 'sparse', and 'lsqr'" % mode).
```

Reply:
thats my fault

Reply:
Fixed

Reply:
Fixed